fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/CastVoteFragment.java

```diff
@@ -0,0 +1,178 @@
+    String [] ballotOptions;
+    Set<String> selectedOptions;
+    int selectedOption;
+    private TextView laoNameText;
+    private TextView voteInText;
+    private TextView electionNameText;
+    private ListView lvBallots;
+    private Button voteButton;
+    private FragmentCastVoteBinding mElectionDisplayFragBinding;
+
+    private LaoDetailViewModel mLaoDetailViewModel;
+
+
+
+    private View.OnClickListener buttonListener = new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            castVote();
+        }
+    };
+
+
+
+    public CastVoteFragment() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static CastVoteFragment newInstance() {
+        return new CastVoteFragment();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        selectedOptions = new HashSet<>();
+         //Inflate the layout for this fragment
+        mElectionDisplayFragBinding =
+                FragmentCastVoteBinding.inflate(inflater,container, false);
+        mLaoDetailViewModel = LaoDetailActivity.obtainViewModel(getActivity());
+        laoNameText = mElectionDisplayFragBinding.castVoteLaoName;
+        mElectionDisplayFragBinding.castVoteVoteInText.setText("Vote in");
+        electionNameText = mElectionDisplayFragBinding.castVoteElectionName;
+
+        //setUp the cast Vote button
+        voteButton = mElectionDisplayFragBinding.castVoteButton;
+        voteButton.setEnabled(false);
+
+        //Getting the Lao Name
+        laoNameText.setText(mLaoDetailViewModel.getCurrentLaoName().getValue());
+
+        //Getting ballot options
+//        setBallotOptions();
+        setDummyBallotOptions();
+
+        //todo get real election name
+        electionNameText.setText("General Election");
+
+        lvBallots = mElectionDisplayFragBinding.castVoteListView;
+
+        //Single options allowed by default
+        numberOfChoices = 1;
+        lvBallots.setChoiceMode(numberOfChoices);
+        ArrayAdapter<String> ballotAdapter = new ArrayAdapter<String>(getActivity(), android.R.layout.simple_list_item_1, ballotOptions);
+
+       // ArrayAdapter<String> ballotAdapter = new ArrayAdapter<String>(getActivity(), android.R.layout.simple_list_item_1, ballotOptions);
+        lvBallots.setAdapter(ballotAdapter);
+
+        lvBallots.setOnItemClickListener(this);
+        voteButton.setOnClickListener(buttonListener);
+        return mElectionDisplayFragBinding.getRoot();
+    }
+
+    private void setBallotOptions(){
```

Comment:
To use this function, you could use the method getBallotOptions() from the election associated. Now, to know which election this fragment is associated with, I guess you would need to retrieve the id corresponding from the ViewModel.